File: src/sys/gtest_v1_runner/tests/echo_test.cc
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <test/placeholders/cpp/fidl.h>
#include <lib/async-loop/cpp/loop.h>
#include <lib/async-loop/default.h>
#include <lib/sys/cpp/service_directory.h>

#include <gtest/gtest.h>

#include "lib/fidl/cpp/binding.h"

TEST(EchoTest, TestEcho) {
  printf("hello echo\n");
  async::Loop loop(&kAsyncLoopConfigAttachToCurrentThread);
  auto service = sys::ServiceDirectory::CreateFromNamespace();
  test::placeholders::EchoSyncPtr echo;
  service->Connect(echo.NewRequest());

  fidl::StringPtr response = "";
  ASSERT_EQ(ZX_OK, echo->EchoString("test string", &response));
  EXPECT_EQ(response, "test string");
}

TEST(EchoTest, TestEcho2) {}

TEST(EchoTest, DISABLED_TestEcho2) {}

File: src/sys/test_runners/legacy_test/test_data/echo_test/echo_test.rs
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

use {fidl_fidl_examples_echo as fecho, fuchsia_component::client::connect_to_protocol};

#[fuchsia::test]
async fn test_echo() {
    const ECHO_STRING: &str = "Hello, world!";
    let echo = connect_to_protocol::<fecho::EchoMarker>().expect("error connecting to echo");
    let out = echo.echo_string(Some(ECHO_STRING)).await.expect("echo_string failed");
    assert_eq!(ECHO_STRING, out.unwrap());
}
